### PR TITLE
feat: add feature to show warning to user if they are already logged in

### PR DESCRIPTION
**Pull request by BRAVO68WEB**
- **feat: added feature to show warning to user if they are already logged in**

### CODE REVIEW

**Comment by BRAVO68WEB** [comment]
closes #11004

**Comment by BRAVO68WEB** [comment]
image

File: cli/login_test.go [hunk 1]
```diff
@@ -442,4 +442,48 @@ func TestLogin(t *testing.T) {
...
+
+		doneChan = make(chan struct{})
+		root, _ = clitest.New(t, "login", "--force-tty", client.URL.String(), "--no-open")
+		pty = ptytest.New(t).Attach(root)
+		go func() {
+			defer close(doneChan)
+			err := root.Run()
+			assert.NoError(t, err)
+		}()
+		pty.ExpectMatch(fmt.Sprintf(" Already Logged In to '%s'!", client.URL.String()))
```
Comment (mafredri) [inline cli/login_test.go:478]: Why do we require a space here?

File: cli/root.go [hunk 2]
```diff
@@ -546,6 +546,50 @@ func (r *RootCmd) InitClient(client *codersdk.Client) serpent.MiddlewareFunc {
...
+				r.token, err = conf.Session().Read()
+				// Even if there isn't a token, we don't care.
+				if err != nil && !os.IsNotExist(err) {
+					return next(inv)
+				}
+			}
+
+			// IF r.token and r.clientUR, exists then print warning "ALready loggedIn"
+			if r.token != "" && r.clientURL.String() != "" {
+				_, _ = fmt.Fprintf(inv.Stdout, Caret+"%s!\n", pretty.Sprint(cliui.DefaultStyles.Warn, "Already Logged In to "+r.clientURL.String()))
```
Comment (mafredri) [inline cli/root.go:581]: What's going on here? This doesn't look syntactically correct, and why "Caret"?

Also, could you make it "You are already logged in to"?

**Review by mafredri — COMMENTED** [review]
Thanks for the PR, could you fix the issues and also make sure the tests pass?

File: cli/root.go (hunk 2 above)
Comment (BRAVO68WEB) [reply cli/root.go:581]: I hope I only needed to change the message to "You are already logged in to $"

**Comment by BRAVO68WEB** [comment]
@mafredri Can you crosscheck now?

File: cli/root.go [hunk 3]
```diff
@@ -546,6 +546,50 @@ func (r *RootCmd) InitClient(client *codersdk.Client) serpent.MiddlewareFunc {
...
+				r.token, err = conf.Session().Read()
+				// Even if there isn't a token, we don't care.
+				if err != nil && !os.IsNotExist(err) {
+					return next(inv)
+				}
+			}
+
+			// IF r.token and r.clientUR, exists then print warning "You are already logged in to $"
+			if r.token != "" && r.clientURL.String() != "" {
+				_, _ = fmt.Fprintf(inv.Stdout, Caret+"%s!\n", pretty.Sprint(cliui.DefaultStyles.Warn, "You are already logged in to "+r.clientURL.String()))
```
[the remark this answers is not in the document]
Comment (matifali) [reply cli/root.go:581]: Yes I agree 💯. Checking token for validity should be a must and if expired should instruct the user to relogin.

[the remark this answers is not in the document]
Comment (BRAVO68WEB) [reply cli/root.go:581]: Guys, give me a final rundown about what should I implement?

[the remark this answers is not in the document]
Comment (BRAVO68WEB) [reply cli/root.go:581]: Noted

**Comment by matifali** [comment]
Hi @BRAVO68WEB , thank you for your contributions! Do you still plan to complete this PR? Looking forward to your updates.

**Comment by BRAVO68WEB** [comment]
i am not about fetch current logged in user.

**Pull request closed without merging**